amundsen_application/static/js/interfaces/Enums.ts

```diff
@@ -7,3 +7,13 @@ export enum FilterType {
   CHECKBOX_SELECT = 'checkboxFilter',
   INPUT_SELECT = 'inputFilter'
 }
+
+export enum SearchType {
+  CLEAR_TERM = 'clearTerm',
+  FILTER = 'filter',
+  INLINE_SEARCH = 'inlineSearch',
+  INLINE_SELECT = 'inlineSelect',
+  LOAD_FROM_URL = 'loadFromUrl',
+  PAGINATION = 'pagination',
+  SEARCH_BAR = 'searchbar',
+}
```

Comment:
I guess we could try to make these slightly more descriptive. 

`clearTerm -> clearSearchBar` (I wasn't initially sure what this referred to)
`filter -> updateFilter`
`searchbar -> submitSearchBar`

Reply:
These are more descriptive but I'm also looking for consistency across the names. I'm going to try a pattern that loosely follows `::verbNoun::` (exception is "inlineSearch/Select").
```
export enum SearchType {
  CLEAR_TERM = 'clearSearchTerm',
  FILTER = 'updateFilter',
  INLINE_SEARCH = 'inlineSearch',
  INLINE_SELECT = 'inlineSelect',
  LOAD_URL = 'loadUrl',
  PAGINATION = 'updatePage',
  SUBMIT_TERM = 'submitSearchTerm',
}
```
Intentionally using `searchTerm` vs. `searchBar` because I consider the "thing" as the information we care about as opposed to the component that updates it.

Reply:
I like to think about these in terms of user action and user intent. 

Also I don't think it's necessary to apply a pattern to these.

Reply:
Unrelated to the content of this PR I'd say we have some more overall guiding principles for frontend logging to flesh out and when the time comes Product should be involved. Something I find tricky to think through is that the purpose of these logs has always been to collect whatever metrics help us make product decisions, which we ourselves do not always know. Another thing I find tricky is that the end result of the actions are also important, but user actions and user intent will not always mirror the end result, because the application logic could be doing any amount of things under the hood and sometimes whats happening under the hood could matter. I don't have the answer or believe there is a clear one, this could very well be a case-by-case kind of issue.

Reply:
I think both types of logging are useful, but for different purposes. For example when a user runs a search, we log API calls for `search_user`, `search_table`, and soon `search_dashboard`. This kind of logging can help track server load, cost, efficiency, etc. This can help us make better engineering decisions. For product related decisions, I think user intent and user actions are the key metrics we should be looking at.